vllm/model_executor/layers/quantization/modelopt.py

```diff
@@ -721,6 +748,80 @@ def __init__(self, quant_config: ModelOptNvFp4Config):
                 raise ValueError("Current platform does not support NVFP4"
                                  " quantization. Please use Blackwell and"
                                  " above.")
+        from vllm.model_executor.layers.fused_moe.cutlass_moe import (
+            cutlass_moe_fp4)
+
+        self.fused_experts = cutlass_moe_fp4
+
+    @property
+    def load_up_proj_weight_first(self) -> bool:
+        # FlashInfer CUTLASS kernel assumes [Up, Gate] Proj as W13
+        return self.allow_flashinfer_cutlass
+
+    def select_experts_impl(self, moe_parallel_config):
+        if not self.allow_flashinfer_cutlass:
+            return
+
+        logger.debug("FlashInferExperts")
```

Comment:
Could this be a debug_once?
```suggestion
        logger.debug_once("Using FlashInferExperts")
```